lib/vanagon/engine/amazon.rb

```diff
@@ -0,0 +1,69 @@
+require 'aws-sdk'
+require 'erb'
+require 'base64'
+require 'vanagon/engine/base'
+
+class Vanagon
+  class Engine
+    class Amazon < Base
+      attr_accessor :ami, :key_name, :userdata, :key, :key_name, :shutdown_behavior
+      attr_accessor :subnet_id, :instance_type, :vpc_id
+
+      def initialize(platform, target = nil)
+        super
+        @name = 'amazon'
+        @platform = platform
+        @ami = @platform.aws_ami
+        @target_user = @platform.target_user
+        @subnet_id = @platform.aws_subnet_id
+        @userdata = @platform.aws_user_data
+        @region = @platform.aws_region.nil? ? 'us-east-1' : @platform.region
```

Comment:
It's probably sufficient to set this like this (I think `@platform.region` is a typo?):

```ruby
@region = @platform.aws_region || 'us-east-1'
```